demo/site/src/util/createSiteUrl.ts

```diff
@@ -0,0 +1,31 @@
+type CreateSiteUrlOptions = {
+    /**
+     * path with leading /
+     *
+     * Sample: "/test/to/my/page"
+     */
+    path: string;
+
+    scope: {
+        language: string;
+    };
+
+    /**
+     * If provided, the anchor will be appended to the resulting URL
+     *
+     * Sample: "my-anchor"
+     */
+    anchor?: string;
+};
+/**
+ * Resolves a path, scope to a reachable path which can be navigated to.
```

Comment:
```suggestion
 * Creates a URL from the provided options.
```

IMO "a reachable path which can be navigated to" says the same thing twice.